### ISSUE TITLE
gem_package does not provide a "live" view of installed gems

### ISSUE THREAD

**Issue by christophermaier** [chef/chef#2361]
The gem_package resource determines which gems are available by using
`Gem::Specification.find_all_by_name` method, which appears to operate
off a snapshot of the gems that are present when the `gem_package`
resource is created. As a result, any changes to the gems in another
part of the Chef run are not reflected.

This can be reproduced with the following recipe:

``` ruby
execute "gem uninstall chefspec --version 4.1.1"

gem_package "chefspec" do
  version "4.1.1"
end
```

If you don't have ChefSpec 4.1.1 installed to begin with, you'll have
it at the end of the run (`gem_package` will run, because the desired
gem wasn't present when the resource was created). However, if you do
have it installed, you won't when the run is done (because the
resource isn't aware that the gem was uninstalled just before hand).

So basically, this recipe will get you what you want, you just need to
converge it twice.
